Sources/VGSShowSDK/Core/APIClient/SatelliteUtils/VGSShowSatelliteUtils.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  VGSShowSatelliteUtils.swift
+//  VGSShowSDK
+//
+//  Created on 22.02.2021.
+//  Copyright © 2021 VGS. All rights reserved.
+//
+
+import Foundation
+
+/// VGSShowSDK utils for satellite.
+internal class VGSShowSatelliteUtils {
+
+	/// Enable assertions on default, disable in unit tests.
+	internal static var isAssertionsEnabled = true
+
+	/// Constants.
+	internal enum Constants {
+
+		/// Valid port numbers. https://en.wikipedia.org/wiki/Port_(computer_networking)#Port_number
+		static let validPortNumbers: ClosedRange<Int> = 1...65535
+
+		/// Valid localhost prefix.
+		static let validLocalHostPrefix = "localhost"
+
+		/// Valid localhost IP address prefix.
+		static let validLocalIPAddressPrefix = "192.168"
+	}
+
+	/// Build URL for satellite.
+	/// - Parameters:
+	///   - environment: `String` object, only `sandbox` is valid.
+	///   - hostname: `String` object, should be valid hostname for satellite.
+	///   - satellitePort: `Int` object, should be valid port.
+	/// - Returns: `URL?`, satellite `URL` or `nil` for invalid configuration.
+	internal static func buildSatelliteURL(with environment: String, hostname: String?, satellitePort: Int) -> URL? {
+
+		// Check satelliteHostname is set.
+		guard let satelliteHostname = hostname else {
+			let errorText = "SATELLITE CONFIGURATION ERROR! HOSTNAME NOT SET! SHOULD BE *http://localhost* or in local IP format  http://192.168.X.X"
+			let event = VGSLogEvent(level: .warning, text: errorText, severityLevel: .error)
+			VGSLogger.shared.forwardLogEvent(event)
+
+			forwardAssertion(with: errorText)
+
+			return nil
+		}
+
+		// Check environment == .sandbox, validate port.
+		guard isEnvironmentValidForSatellite(environment), isSatellitePortValid(satellitePort) else {
+			return nil
+		}
+
+		// Check hostname is valid for satellite.
+		guard isSatelliteHostnameValid(satelliteHostname) else {
+			return nil
```

Comment:
Should we throw warning here?

Reply:
No. We throw warnings in the internal methods.
Please check `isSatelliteHostnameValid`